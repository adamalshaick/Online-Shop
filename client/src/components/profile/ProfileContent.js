import React from "react";
import PropTypes from "prop-types";
import AddReview from "../reviews/AddReview";
import { connect } from "react-redux";
import TableCell from "@material-ui/core/TableCell";
import TableRow from "@material-ui/core/TableRow";

export const ProfileContent = ({ user, auth }) => {
  return (
    <>
      <div className="text-center">
        <div className="mt-3">
          <TableRow>
            <TableCell>
              <strong>User's Name</strong>
            </TableCell>
            <TableCell align="center">{user.name}</TableCell>
          </TableRow>
          <TableRow>
            <TableCell>
              <strong>User's Location</strong>
            </TableCell>
            <TableCell align="center">{user.location}</TableCell>
          </TableRow>
          <TableRow>
            <TableCell>
              <strong>User's Bio</strong>
            </TableCell>
            <TableCell align="center">{user.bio}</TableCell>
          </TableRow>
        </div>
      </div>
      {user._id === auth.user.id ? null : <AddReview user={user} />}
    </>
  );
};

ProfileContent.propTypes = {
  user: PropTypes.object.isRequired,
  deleteAccount: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(mapStateToProps)(ProfileContent);
