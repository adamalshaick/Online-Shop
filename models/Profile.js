const mongoose = require("mongoose");
const Schema = mongoose.Schema;

// Create Schema
const ProfileSchema = new Schema({
  user: {
    type: Schema.Types.ObjectId,
    ref: "users"
  },
  handle: {
    type: String,
    required: true,
    max: 40
  },

  bio: {
    type: String
  },

  location: {
    type: String
  },

  profileImage: {
    type: String
  },

  cart: {
    items: [
      {
        user: {
          type: Schema.Types.ObjectId,
          ref: "users"
        },
        item: {
          type: Schema.Types.ObjectId,
          ref: "items"
        },
        text: {
          type: String,
          required: true
        },
        price: {
          type: String,
          required: true
        },
        title: {
          type: String,
          required: true
        },
        itemImage: {
          type: String,
          required: true
        }
      }
    ]
  },

  itemsBought: [
    {
      id: {
        type: Schema.Types.ObjectId,
        ref: "items"
      }
    }
  ],

  itemsSold: [
    {
      id: {
        type: Schema.Types.ObjectId,
        ref: "items"
      }
    }
  ],

  date: {
    type: Date,
    default: Date.now
  }
});

module.exports = Profile = mongoose.model("profile", ProfileSchema);
