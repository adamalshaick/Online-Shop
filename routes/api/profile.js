const express = require("express");
const router = express.Router();
const passport = require("passport");

// Controllers
const profileController = require("../../controllers/profileController");

// @route GET api/profile
// @desc Get current users profile
// @access Private
router.get(
  "/",
  passport.authenticate("jwt", { session: false }),
  (req, res) => {
    profileController.getProfile(req, res);
  }
);

// @route GET api/profile/all
// @desc Get all profiles
// @access Public
router.get("/all", (req, res) => {
  profileController.getAllProfiles(req, res);
});

// @route GET api/profile/handle/:handle
// @desc Get profile by handle
// @access Public
router.get("/handle/:handle", (req, res) => {
  profileController.getProfileByHandle(req, res);
});

// @route GET api/profile/user/:user_id
// @desc Get profile by user ID
// @access Public

router.get("/user/:user_id", (req, res) => {
  profileController.getProfileById(req, res);
});

// @route POST api/profile
// @desc Create or edit user profile
// @access Private
router.post(
  "/",
  passport.authenticate("jwt", { session: false }),
  (req, res) => {
    profileController.createProfile(req, res);
  }
);

// @route GET api/profiles
// @desc Get profiles
// @access Public
router.get("/profiles", (req, res) => {
  profileController.getProfiles(req, res);
});

// @route   DELETE api/profile
// @desc    Delete user and profile
// @access  Private
router.delete(
  "/",
  passport.authenticate("jwt", { session: false }),
  (req, res) => {
    deleteProfile(req, res);
  }
);

module.exports = router;
