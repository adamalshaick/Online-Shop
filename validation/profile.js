const Validator = require("validator");
const isEmpty = require("./is-empty");

module.exports = function validateProfileInput(data, uploadError) {
  let errors = {};

  data.handle = !isEmpty(data.handle) ? data.handle : "";

  if (!Validator.isLength(data.handle, { min: 2, max: 40 })) {
    errors.handle = "Handle needs to be between 2 and 40 characters";
  }

  if (Validator.isEmpty(data.handle)) {
    errors.handle = "Username is required";
  }

  if (uploadError) {
    errors.file = uploadError;
  }

  return {
    errors,
    isValid: isEmpty(errors)
  };
};
