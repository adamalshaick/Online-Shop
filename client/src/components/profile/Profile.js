import React from "react";
import PropTypes from "prop-types";
import ProfileContent from "./ProfileContent";
import Reviews from "../reviews/Reviews";
import fetchUserById from "../common/hoc/fetchUserById";
import { SecondaryHeader } from "../common/styles/Header";

export const Profile = ({ user }) => {
  return (
    <div className="entry container mt-md-5">
      <div className="row">
        <section className="col-6">
          <SecondaryHeader className="text-center">
            Profile Info <hr />
          </SecondaryHeader>
          <ProfileContent user={user.user} />
        </section>
        <section className="col-6">
          <SecondaryHeader className="text-center">
            Reviews <hr />
          </SecondaryHeader>
          <Reviews user={user.user} />
        </section>
      </div>
    </div>
  );
};

Profile.propTypes = {
  user: PropTypes.object.isRequired
};

export default fetchUserById(Profile);
