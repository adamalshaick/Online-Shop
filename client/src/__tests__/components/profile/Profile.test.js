import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Profile } from "../../../components/profile/Profile";

const Props = {
  user: { user: {} }
};

const wrapper = shallow(<Profile {...Props} />);
describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
