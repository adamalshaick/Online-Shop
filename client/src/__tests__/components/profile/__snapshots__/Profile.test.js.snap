// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<div
  className="entry container mt-md-5"
>
  <div
    className="row"
  >
    <section
      className="col-6"
    >
      <styled.header
        className="text-center"
      >
        Profile Info 
        <hr />
      </styled.header>
      <Connect(ProfileContent)
        user={Object {}}
      />
    </section>
    <section
      className="col-6"
    >
      <styled.header
        className="text-center"
      >
        Reviews 
        <hr />
      </styled.header>
      <Connect(ComposedComponent)
        user={Object {}}
      />
    </section>
  </div>
</div>
`;
